baseplate/context/memcache/__init__.py

```diff
@@ -107,31 +107,147 @@ def __init__(self, context_name, server_span, pooled_client):
         self.server_span = server_span
         self.pooled_client = pooled_client
 
-    #pylint: disable=no-self-argument
-    def monitored(fn_name):
-        def proxy_with_instrumentation(self, *a, **kw):
-            trace_name = "{}.{}".format(self.context_name, fn_name)
-            with self.server_span.make_child(trace_name):
-                method = getattr(self.pooled_client, fn_name)
-                return method(*a, **kw)
-        return proxy_with_instrumentation
-
-    close = monitored('close')
-    set = monitored('set')
-    set_many = monitored('set_many')
-    replace = monitored('replace')
-    append = monitored('append')
-    prepend = monitored('prepend')
-    cas = monitored('cas')
-    get = monitored('get')
-    get_many = monitored('get_many')
-    gets_many = monitored('gets_many')
-    delete = monitored('delete')
-    delete_many = monitored('delete_many')
-    add = monitored('add')
-    incr = monitored('incr')
-    decr = monitored('decr')
-    touch = monitored('touch')
-    stats = monitored('stats')
-    flush_all = monitored('flush_all')
-    quit = monitored('quit')
+    def close(self):
+        with self._get_span("close"):
+            return self.pooled_client.close()
+
+    def set(self, key, value, expire=0, noreply=None):
+        with self._get_span("set") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
+            return self.pooled_client.set(
+                key, value, expire=expire, noreply=noreply)
+
+    def set_many(self, values, expire=0, noreply=None):
+        with self._get_span("set_many") as span:
+            span.set_tag("key_count", len(values))
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            return self.pooled_client.set_many(
+                values, expire=expire, noreply=noreply)
+
+    def replace(self, key, value, expire=0, noreply=None):
+        with self._get_span("replace") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
+            return self.pooled_client.replace(
+                key, value, expire=expire, noreply=noreply)
+
+    def append(self, key, value, expire=0, noreply=None):
+        with self._get_span("append") as span:
+            span.set_tag("key", key)
+            span.set_tag("expire", expire)
+            span.set_tag("noreply", noreply)
+            span.set_tag("value", value)
```

Comment:
In the context of the meeting we had the other day, I think we should avoid sending `value`s to the tracing system as they're quite likely to have user-data in them. (And regardless may be large binary blobs)